superset/assets/src/SqlLab/components/SqlEditorLeftBar.jsx

```diff
@@ -41,12 +41,10 @@ class SqlEditorLeftBar extends React.PureComponent {
   onDatabaseChange(db, force) {
     const val = db ? db.value : null;
     this.setState({ schemaOptions: [] });
+    this.setState({ tableOptions: [] });
     this.props.actions.queryEditorSetSchema(this.props.queryEditor, null);
     this.props.actions.queryEditorSetDb(this.props.queryEditor, val);
-    if (!(db)) {
-      this.setState({ tableOptions: [] });
-    } else {
-      this.fetchTables(val, this.props.queryEditor.schema);
```

Comment:
fetchTables here makes no sense because the database has been changed but `this.props.queryEditor.schema` is still the schema selected for the previously selected db. 
And it is set null on line 44:
`this.props.actions.queryEditorSetSchema(this.props.queryEditor, null);`